### DISCUSSION TITLE
Using a dedicated server to host the HOMES

### DISCUSSION THREAD

**Discussion by maccadia** [ltsp/ltsp#573] [Q&A] [answered]
Hi,

I would like to use a dedicated server for hosting the users homes. I have a LTSP server and a HOME server accessible via SSH.

My setup works with the default `ltsp.conf` from `/usr/share/ltsp/common/ltsp/ltsp.conf` i.e. using the LTSP server to host the HOMES.

When I add the variable `SSH_SERVER=[IP_OF_LTSP_SERVER]` in the `[clients]` section. GDM login hangs forever (spinning). Same thing with the IP of the HOME server.

When I add the variable `SERVER=[IP_OF_HOME_SERVER]` in the `[clients]` section. I get _"pamltsp : failed to mount home via SSHFS. Connection reset by peer."_

I tried to connect to the HOME server via ssh from the LTSP client and it works.

I suspect that I need to add the HOME server's pubkey to the known_hosts but how ?

**Comment by alkisg** [comment, marked as answer]
Put them in /etc/ltsp/ssh_known_hosts, then run `ltsp initrd`.

**Comment by maccadia** [comment]
Thanks, it works.

I used your command to create the known_hosts file :
```
# on the HOME server
sed "s/^/server /" /etc/ssh/ssh_host_*_key.pub
```

I then needed to add some `sshfs_params` since the uid is different on the LTSP client and the HOME server. I replaced [this](https://github.com/ltsp/ltsp/blob/7961c6d90172f13421c407d3ac42754b80ff0557/ltsp/client/login/pamltsp#L189) with : 
```
sshfs_params="password_stdin,allow_other,default_permissions,idmap=user,uid=$(id -u $PAM_USER)"
```

**Comment by alkisg** [comment]
I think you should just install ltsp on the home server, and run ltsp initrd, and get ltsp.img, and put it on the ltsp server.
The ltsp package by itself has no dependencies, it won't disrupt anything.

**Reply by maccadia** [reply]
Thank you for the tip. In my case it's not possible because the authentication system on the home server (LDAP) is not the same as the one on the ltsp clients (because the LDAP admins don't want to give access to the central LDAP from the clients).
